plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -298,38 +349,74 @@ open class DokkaDescriptorVisitor( // TODO: close this class and make it private
+        getContributedDescriptors(DescriptorKindFilter.PACKAGES) { true }
+            .filterIsInstance<PackageFragmentDescriptor>()
+            .map { visitPackageFragmentDescriptor(it, parent) }
+
+    private fun MemberScope.enumEntries(parent: DRIWithPlatformInfo): List<EnumEntry> =
+        this.getContributedDescriptors(DescriptorKindFilter.CLASSIFIERS) { true }
+            .filterIsInstance<ClassDescriptor>()
+            .map { enumEntryDescriptor(it, parent) }
 
-        return BasePlatformInfo(docHeader, listOf(platformData))
-    }
 
-    private fun ClassDescriptor.resolveClassDescriptionData(): ClassPlatformInfo {
-        return ClassPlatformInfo(resolveDescriptorData(),
-            (getSuperInterfaces() + getAllSuperclassesWithoutAny()).map { DRI.from(it) })
+    private fun DeclarationDescriptor.resolveDescriptorData(platformData: PlatformData): PlatformDependent<DocumentationNode> =
+        PlatformDependent.from(platformData, getDocumentation())
+
+    private fun ClassDescriptor.resolveClassDescriptionData(platformData: PlatformData): ClassInfo {
+        return ClassInfo(
+            (getSuperInterfaces() + getAllSuperclassesWithoutAny()).map { DRI.from(it) },
+            resolveDescriptorData(platformData)
+        )
     }
 
-    private fun PlatformInfo?.filterTagWrappers(
-        types: List<KClass<out TagWrapper>>,
-        name: String? = null
-    ): PlatformInfo? {
-        if (this == null)
-            return null
-        return BasePlatformInfo(
-            DocumentationNode(
-                this.documentationNode.children.filter { it::class in types && (it as? NamedTagWrapper)?.name == name }
-            ),
-            this.platformData
+    private fun TypeParameterDescriptor.toTypeParameter() =
+        TypeParameter(
+            DRI.from(this),
+            fqNameSafe.asString(),
+            PlatformDependent.from(platformData, getDocumentation()),
+            upperBounds.map { it.toProjection() },
+            listOf(platformData)
+        )
+
+    private fun KotlinType.toProjection(): Projection = when (constructor.declarationDescriptor) {
+        is TypeParameterDescriptor -> Projection.OtherParameter(fqName.toString()).let {
+            if (isMarkedNullable) Projection.Nullable(it) else it
+        }
+        else -> Projection.TypeConstructor(
+            DRI.from(constructor.declarationDescriptor!!), // TODO: remove '!!'
+            arguments.map { it.toProjection() }
         )
     }
 
-    private fun List<PlatformInfo>.filterTagWrappers(
-        types: List<KClass<out TagWrapper>>,
-        name: String? = null
-    ): List<PlatformInfo> =
-        this.map { it.filterTagWrappers(types, name)!! }
-}
+    private fun TypeProjection.toProjection(): Projection =
+        if (isStarProjection) Projection.Star else fromPossiblyNullable(type)
 
+    private fun fromPossiblyNullable(t: KotlinType): Projection =
```

Comment:
You are using this function once, while you use `let { if (t.isMarkedNullable) Projection.Nullable(it) else it }` twice.  Maybe it's better to functionize this instead?